pkg/manager/member/pvc_resizer.go

```diff
@@ -385,132 +375,284 @@ func (p *pvcResizer) updateVolumeStatus(ctx *componentVolumeContext) {
+		}
+
+		// remove leader eviction ann from reized pod
+		for _, podVolume := range ctx.actualPodVolumes {
+			pod := podVolume.pod
+
+			if pod.Name == resizePod.Name {
+				break
 			}
-			_, err = p.deps.KubeClientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Patch(context.TODO(), pvc.Name, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+
+			updated, err := updateResizeAnnForTiKVPod(p.deps.KubeClientset, false, pod)
 			if err != nil {
-				errors = append(errors, err)
-				continue
+				return err
+			}
+			if updated {
+				klog.Infof("%s: remove leader eviction annotation from pod %s/%s", logPrefix, pod.Namespace, pod.Name)
 			}
-
-			klog.V(2).Infof("PVC %q of %q storage request is updated from %s to %s",
-				pvcID, id, currentRequest.String(), quantityInSpec.String())
 		}
 
-		if len(errors) > 0 {
-			return errutil.NewAggregate(errors)
+		// add leader eviction ann to the pod and wait the leader count to be 0
+		updated, err := updateResizeAnnForTiKVPod(p.deps.KubeClientset, true, resizePod)
+		if err != nil {
+			return err
+		}
+		if updated {
+			klog.Infof("%s: add leader eviction annotation to pod", logPrefix)
+		}
+		for _, store := range tc.Status.TiKV.Stores {
+			if store.PodName == resizePod.Name {
+				if store.LeaderCount == 0 {
+					klog.V(4).Infof("%s: leader count of store %s become 0", logPrefix, store.ID)
+					return nil
+				} else {
+					return controller.RequeueErrorf("%s: wait for leader count of store %s to be 0", logPrefix, store.ID)
+				}
+			}
 		}
 
-		if !allResized {
-			klog.Infof("Skip to resize other PVCs of %q: wait all PVCs in Pod %s/%s to be resized",
-				id, podVolume.pod.Namespace, podVolume.pod.Name)
-			break
+		return fmt.Errorf("%s: can't find store in tc", logPrefix)
+	}
+
+	return nil
+}
+
+func (p *pvcResizer) beginResize(ctx *componentVolumeContext) error {
+	ctx.status.SetCondition(metav1.Condition{
+		Type:    v1alpha1.ComponentVolumeResizing,
+		Status:  metav1.ConditionTrue,
+		Reason:  "BeginResizing",
+		Message: "Set resizing condition to begin resizing",
+	})
+	klog.Infof("begin resizing for %s: set resizing condition", ctx.ComponentID())
+	return controller.RequeueErrorf("set condition before resizing volumes for %s", ctx.ComponentID())
+}
+
+func (p *pvcResizer) endResize(ctx *componentVolumeContext) error {
+	switch ctx.status.GetMemberType() {
+	case v1alpha1.TiKVMemberType:
+		// ensure all eviction annotations are removed
+		for _, podVolume := range ctx.actualPodVolumes {
+			updated, err := updateResizeAnnForTiKVPod(p.deps.KubeClientset, false, podVolume.pod)
+			if err != nil {
+				return fmt.Errorf("remove leader eviction annotation from pod failed: %s", err)
+			}
+			if updated {
+				klog.Infof("end resizing for %s: remove leader eviction annotation from pod %s/%s",
+					ctx.ComponentID(), podVolume.pod.Namespace, podVolume.pod.Name)
+			}
 		}
 	}
 
+	ctx.status.RemoveCondition(v1alpha1.ComponentVolumeResizing)
```

Comment:
Is it better to just set the condition to false instead of removing it?

Reply:
Then we also have the timestamp (last transition timestamp) when the last resizing is done.